codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/SymbolVisitor.java

```diff
@@ -233,22 +233,22 @@ public Symbol setShape(SetShape shape) {
 
     private Symbol createCollectionSymbol(CollectionShape shape) {
         Symbol reference = toSymbol(shape.getMember());
-        return SymbolUtils.createValueSymbolBuilder(shape, reference.getName(), reference.getNamespace())
+        return SymbolUtils.createValueSymbolBuilder(shape, shape.getId().getName())
```

Comment:
Why this particular change? Also, you'll want to use `getDefaultShapeName`, which handles casing.

Reply:
Well the naming is actually meaningless as the `GO_ELEMENT_TYPE` storing the symbol will convey the actual name to be used when it is further unraveled since this is a slice. Honestly the name here could be empty string or null. I chose to use the name of the actual collection shape in case it were to get incorrectly rendered into code.

Reply:
This is probably fine then, if you use the previously mentioned function and add the comment. That name theoretically be real if we were to generate a type alias. The code won't compile if you mess it up, but you'll know where to look since you've got the shape name.